File: app/src/main/java/com/juawapps/newsspread/di/AppModuleRepository.java
package com.juawapps.newsspread.di;

import com.juawapps.newsspread.data.ArticlesRepository;
import com.juawapps.newsspread.data.api.NewsapiService;
import com.juawapps.newsspread.data.db.ArticleDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Repository module
 */
@Module
public class AppModuleRepository {

    @Singleton
    @Provides
    ArticlesRepository provideArticleRepository(NewsapiService newsapiService,
                                                ArticleDao articleDao) {
        return new ArticlesRepository(newsapiService, articleDao);
    }
}

File: app/src/main/java/com/juawapps/newsspread/di/AppModuleRepository.kt
package com.juawapps.newsspread.di

import com.juawapps.newsspread.data.ArticlesRepository
import com.juawapps.newsspread.data.api.NewsapiService
import com.juawapps.newsspread.data.db.ArticleDao
import com.juawapps.newsspread.utils.AppExecutors

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Repository module
 */
@Module
class AppModuleRepository {

    @Singleton
    @Provides
    internal fun provideArticleRepository(newsapiService: NewsapiService,
                                          articleDao: ArticleDao,
                                          appExecutors: AppExecutors): ArticlesRepository {
        return ArticlesRepository(newsapiService, articleDao, appExecutors)
    }
}
